SofaKernel/modules/SofaBaseTopology/GridTopology.h

```diff
@@ -195,6 +210,9 @@ using MeshTopology::getHexahedron;
     Data<bool> d_computeHexaList, d_computeQuadList, d_computeEdgeList, d_computePointList;
     /// Data bool to set option to compute texcoords
     Data<bool> d_createTexCoords;
+
+    /// Enum storing the actual dimension of this grid
+    Grid_dimension m_gridDim;
```

Comment:
Up to my understanding, GridTopology have dimension once they are initialized...this is property of the topology that can be  recomputed from the nx,ny,nz each time we need it or stored. I have not opinion on that as long as there is function to retrieve the dimension.